mdn/developer-portal
# 1009: Add support for conditionally including Topic records in topics lists

This changeset addresses the need to keep using `Topic`s for categorisation, but not to include some of them in the "topics block" listing layouts on the homepage and (hidden) /topics/ page

 (Resolves #1009)

## Key changes:

* Updates Topic model to have new show_in_list_views field, default True.
* Adds new schema migration for the above
* Updates ORM calls made by homepage and topics pages to filter by this new show_in_list_views field
* Added tests

**Admin UI change**
<img width="746" alt="Screenshot 2019-12-30 at 22 26 58" src="https://user-images.githubusercontent.com/101457/71604322-1ef9ea00-2b59-11ea-92fa-2c6823385476.png">

**See the CSS topic present**
<img width="1111" alt="Screenshot 2019-12-30 at 22 29 44" src="https://user-images.githubusercontent.com/101457/71604323-1ef9ea00-2b59-11ea-82f3-864acfb52f8a.png">

**...and no longer present**
<img width="1158" alt="Screenshot 2019-12-30 at 22 31 25" src="https://user-images.githubusercontent.com/101457/71604324-1ef9ea00-2b59-11ea-9ee3-8ec54757f4e9.png">



## How to test

- on https://developer-portal.dev.mdn.mozit.cloud/admin/ edit one or more Topics to hide them from list view and re-publish them -- I've already hidden CSS fro the list views
- confirm they are no longer shown in the list on the homepage
